### PR TITLE
Ported NPE from v2

### PR DESCRIPTION
**Pull request by kushagraThapar**
This PR is a port of v2 fix: https://github.com/Azure/azure-cosmosdb-java/pull/267

### CODE REVIEW

File: sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/directconnectivity/AddressResolver.java [hunk 1]
```diff
@@ -548,6 +551,7 @@ private ResolutionResult handleRangeAddressResolutionFailure(
             throw BridgeInternal.setResourceAddress(new PartitionKeyRangeGoneException(errorMessage), request.getResourceAddress());
         }
 
+        logger.debug("handleRangeAddressResolutionFailure returns null");
         return null;
```
Comment (moderakh) [inline sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/directconnectivity/AddressResolver.java]: is this return null correct?

Comment (kushagraThapar) [reply sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/directconnectivity/AddressResolver.java]: Yes, this null return is correct :)

**Comment by mitchdenny** [comment]
Merging after conversation with @kushagraThapar

**Pull request merged**